### PR TITLE
the default recipe no longer should be combined...

### PR DESCRIPTION
**Pull request by jspiewak**
with replicaset or shard, so it does not need to check if the replicaset recipe is loaded

See #236

### CODE REVIEW

**Comment by jamesonjlee** [comment]
I was about to merge then, I realize actual issue is that the unless checks are partially wrong.  They should all be an `if node['mongodb']['is_...']` checks.

**Comment by jspiewak** [comment]
In the case of default.rb, there doesn't need to be a check. The only time it should be used is if you are configuring a standalone instance.

Right, now the only check left in the recipes is in replicaset.rb because it might be combined with shard.rb. And yes, that is as you say.

Do you want me to change that as part of this PR?

**Comment by jamesonjlee** [comment]
yes, please? combine all the unless/recipe logic into the `is_...` type and we can remove this in the default recipe.

**Comment by jspiewak** [comment]
I need to update some thing on my home computer in order to run the tests. Will get to that after breakfast and coffee. But this is what I am thinking....

**Comment by jamesonjlee** [comment]
yep something like this.

**Comment by jspiewak** [comment]
Had a test failure at home and then got distracted by other projects.
Back at work today, I managed to get the same failure on configserver-centos-65:

```
I, [2014-02-18T15:53:27.225656 #1881]  INFO -- configserver-centos-65: Finished setting up <configserver-centos-65> (0m51.38s).
I, [2014-02-18T15:53:27.225822 #1881]  INFO -- configserver-centos-65: -----> Verifying <configserver-centos-65>...
I, [2014-02-18T15:53:27.604666 #1881]  INFO -- configserver-centos-65: Suite path directory /tmp/busser/suites does not exist, skipping.
I, [2014-02-18T15:53:27.776607 #1881]  INFO -- configserver-centos-65: Uploading /tmp/busser/suites/bats/default.bats (mode=0644)
I, [2014-02-18T15:53:28.254461 #1881]  INFO -- configserver-centos-65: -----> Running bats test suite
I, [2014-02-18T15:53:28.307730 #1881]  INFO -- configserver-centos-65:  ✗ starts configserver
I, [2014-02-18T15:53:28.307790 #1881]  INFO -- configserver-centos-65:    (in test file /tmp/busser/suites/bats/default.bats, line 6)
I, [2014-02-18T15:53:28.307856 #1881]  INFO -- configserver-centos-65: 
I, [2014-02-18T15:53:28.307877 #1881]  INFO -- configserver-centos-65: 1 test, 1 failure
I, [2014-02-18T15:53:28.307896 #1881]  INFO -- configserver-centos-65: Command [/tmp/busser/vendor/bats/bin/bats /tmp/busser/suites/bats] exit code was 1
E, [2014-02-18T15:53:28.316836 #1881] ERROR -- configserver-centos-65: Verify failed on instance <configserver-centos-65>.
E, [2014-02-18T15:53:28.317004 #1881] ERROR -- configserver-centos-65: ------Exception-------
E, [2014-02-18T15:53:28.317029 #1881] ERROR -- configserver-centos-65: Class: Kitchen::ActionFailed
E, [2014-02-18T15:53:28.317086 #1881] ERROR -- configserver-centos-65: Message: SSH exited (1) for command: [sh -c 'BUSSER_ROOT="/tmp/busser" GEM_HOME="/tmp/busser/gems" GEM_PATH="/tmp/busser/gems" GEM_CACHE="/tmp/busser/gems/cache" ; export BUSSER_ROOT GEM_HOME GEM_PATH GEM_CACHE; sudo -E /tmp/busser/bin/busser test']
E, [2014-02-18T15:53:28.317107 #1881] ERROR -- configserver-centos-65: ---Nested Exception---
E, [2014-02-18T15:53:28.317127 #1881] ERROR -- configserver-centos-65: Class: Kitchen::SSHFailed
E, [2014-02-18T15:53:28.317146 #1881] ERROR -- configserver-centos-65: Message: SSH exited (1) for command: [sh -c 'BUSSER_ROOT="/tmp/busser" GEM_HOME="/tmp/busser/gems" GEM_PATH="/tmp/busser/gems" GEM_CACHE="/tmp/busser/gems/cache" ; export BUSSER_ROOT GEM_HOME GEM_PATH GEM_CACHE; sudo -E /tmp/busser/bin/busser test']
E, [2014-02-18T15:53:28.317168 #1881] ERROR -- configserver-centos-65: ------Backtrace-------
E, [2014-02-18T15:53:28.317187 #1881] ERROR -- configserver-centos-65: /Users/jspiewak/.rbenv/versions/2.1.0/lib/ruby/gems/2.1.0/gems/test-kitchen-1.1.1/lib/kitchen/driver/ssh_base.rb:111:in `rescue in run_remote'
E, [2014-02-18T15:53:28.317207 #1881] ERROR -- configserver-centos-65: /Users/jspiewak/.rbenv/versions/2.1.0/lib/ruby/gems/2.1.0/gems/test-kitchen-1.1.1/lib/kitchen/driver/ssh_base.rb:107:in `run_remote'
```

Looks like it fails on master too. So I think this is good to go.

**Comment by jspiewak** [comment]
Updated PR for latest master.

File: definitions/mongodb.rb [hunk 1]
```diff
@@ -234,10 +235,8 @@
 
     ruby_block 'config_sharding' do
       block do
-        if new_resource.type == 'mongos'
```
Comment (jamesonjlee) [inline definitions/mongodb.rb:237]: was this check moot?

Comment (jspiewak) [inline definitions/mongodb.rb:237]: yep, the block only exists if the type is mongos

**Pull request merged**